client/context/loadbalancing.go

```diff
@@ -0,0 +1,46 @@
+package context
+
+import (
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	"strings"
+	"sync"
+	"github.com/pkg/errors"
+)
+
+// ClientManager is a manager of a set of rpc clients to full nodes.
+// This manager can do load balancing upon these rpc clients.
+type ClientManager struct {
+	clients []rpcclient.Client
+	currentIndex int
+	mutex sync.RWMutex
```

Comment:
I think it'll suffice to just have a write only mutex here (I don't see any logic that obtains a read-lock).

Reply:
Do you mean use sync.Mutex instead?